hwy/ops/generic_ops-inl.h

```diff
@@ -2808,13 +2808,35 @@ HWY_API VFromD<D> GatherIndexN(D d, const T* HWY_RESTRICT base,
   }
   return v;
 }
+
+template <class D, typename T = TFromD<D>>
+HWY_API VFromD<D> GatherIndexN(VFromD<D> no, D d, const T* HWY_RESTRICT base,
+                               VFromD<RebindToSigned<D>> index,
+                               const size_t max_lanes_to_load) {
+  const RebindToSigned<D> di;
+  using TI = TFromD<decltype(di)>;
+  static_assert(sizeof(T) == sizeof(TI), "Index/lane size must match");
+
+  VFromD<D> v = no;
+  for (size_t i = 0; i < MaxLanes(d); ++i) {
```

Comment:
Maybe `i < HWY_MIN(MaxLanes(d), max_lanes_to_load)`? Or break after the if condition is false?